lib/app/services/search_service.dart

```diff
@@ -1,21 +1,21 @@
+import 'package:favoritism_communication/app/dummy_data/chat_dummy_data.dart';
 import 'package:get/get.dart';
+import 'package:favoritism_communication/app/components/organisms/organisms.dart';
 
 class SearchService extends GetxService {
   late List<String> targetUsers;
   RxList<String> results = <String>[].obs;
   static const _searchIntervalMilliSecond = 1000; //1秒
```

Comment:
```suggestion
　　　RxString target = ''.obs;
```
waitInputAndSearchメソッドをなくして簡単にしたいので、
手順追って修正お願いします(リファクタ①)

Reply:
リファクタ②
このメソッドの削除お願いします
```
void waitInputAndSearch(String? text) {
    //前回テキスト変更時Datetimeを更新
    DateTime lastChangedDatetime = DateTime.now();
    //一定時間間隔(_searchIntervalMilliSecond)以内にテキストの変更を検知した場合は_prefixSearchの実行を待つ
    Future.delayed(
      const Duration(milliseconds: _searchIntervalMilliSecond),
      () {
        final DateTime nowDatetime = DateTime.now();
        if (nowDatetime.difference(lastChangedDatetime).inMilliseconds >
            _searchIntervalMilliSecond) {
          results.value = _prefixSearch(text);
        }
      },
    );
  }
```

Reply:
リファクタ③
_prefixSearch(String? text) メソッドを以下のように修正します

```
void prefixSearch(String? text) {
    List<String> searchedResults = [];

    if (text == null || text == '') {
      //処理なし：searchedResultListは空のまま
    } else {
      searchedResults = targetUsers
          .where(
              (element) => element.toLowerCase().startsWith(text.toLowerCase()))
          .toList();
    }
    results.assignAll(searchedResults);
  }
```

Reply:
@king-kazu39 
ご指摘の通り修正いたしました。